Index: GDL.pm
===================================================================
--- GDL.pm	(revision 255)
+++ GDL.pm	(working copy)
@@ -172,7 +172,7 @@
     }
 
 
-    foreach $cl (sort keys %{$classes})
+    foreach $cl (sort {classcmp($a, $b)} keys %{$classes})
     {
         $fh->print("c$cl = ($glyphs->[$classes->{$cl}[0]]{'name'}");
         for ($i = 1; $i <= $#{$classes->{$cl}}; $i++)
@@ -180,7 +180,7 @@
         $fh->print(");\n\n");
     }
 
-    foreach $cl (sort keys %{$ligclasses})
+    foreach $cl (sort {classcmp($a, $b)} keys %{$ligclasses})
     {
         $fh->print("cl$cl = ($glyphs->[$ligclasses->{$cl}[0]]{'name'}");
         for ($i = 1; $i <= $#{$ligclasses->{$cl}}; $i++)
@@ -191,6 +191,15 @@
     $self;
 }
 
+sub classcmp
+{
+    my ($x, $y) = @_;
+    my ($v, $w) = ($x, $y);
+    $v =~ s/^no_//o;
+    $w =~ s/^no_//o;
+    return ($v cmp $w || $x cmp $y);
+}
+
 sub endtable
 {
     my ($self, $fh) = @_;
Index: AP.pm
===================================================================
--- AP.pm	(revision 259)
+++ AP.pm	(working copy)
@@ -508,12 +508,14 @@
     {
         foreach $name (split('/', $glyph->{'post'}))
         {
+            my ($gnum) = $glyph->{'gnum'};
             while ($name =~ m/\.([^_.]+)$/o)
             {
                 my ($base, $ext) = ($` , $1);    #` make editor happy
                 last unless ($i = $namemap{$base});
-                push (@{$classes{$ext}}, $glyph->{'gnum'});
-                push (@{$classes{"no_$ext"}}, $self->{'glyphs'}[$i]{'gnum'});
+                push (@{$classes{$ext}}, $gnum);
+                $gnum = $self->{'glyphs'}[$i]{'gnum'};
+                push (@{$classes{"no_$ext"}}, $gnum);
                 $name = $base;
             }
         }
